heron/instance/src/python/instance/st_heron_instance.py

```diff
@@ -0,0 +1,269 @@
+# Copyright 2016 Twitter. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+'''module for single-thread Heron Instance in python'''
+import collections
+import logging
+import os
+import sys
+import traceback
+import signal
+import yaml
+
+from heron.common.src.python.basics import GatewayLooper
+from heron.common.src.python.utils import log
+from heron.common.src.python.utils.metrics import GatewayMetrics, MetricsCollector
+from heron.common.src.python.utils.misc import HeronCommunicator
+from heron.common.src.python.network import create_socket_options
+
+from heron.proto import physical_plan_pb2, stmgr_pb2
+from heron.instance.src.python.network import MetricsManagerClient, SingleThreadStmgrClient
+from heron.instance.src.python.basics import SpoutInstance, BoltInstance
+
+import heron.common.src.python.constants as constants
+
+Log = log.Log
+AssignedInstance = collections.namedtuple('AssignedInstance', 'is_spout, protobuf, py_class')
+
+# pylint: disable=too-many-instance-attributes
+class SingleThreadHeronInstance(object):
+  """SingleThreadHeronInstance is an implementation of Heron Instance in python"""
+  STREAM_MGR_HOST = "127.0.0.1"
+  METRICS_MGR_HOST = "127.0.0.1"
+  def __init__(self, topology_name, topology_id, instance,
+               stream_port, metrics_port, topo_pex_file_path, sys_config):
+    # Basic information about this heron instance
+    self.topology_name = topology_name
+    self.topology_id = topology_id
+    self.instance = instance
+    self.stream_port = stream_port
+    self.metrics_port = metrics_port
+    self.topo_pex_file_abs_path = os.path.abspath(topo_pex_file_path)
+    self.sys_config = sys_config
+
+    self.in_stream = HeronCommunicator(producer_cb=None, consumer_cb=None)
+    self.out_stream = HeronCommunicator(producer_cb=None, consumer_cb=None)
+
+    self.socket_map = dict()
+    self.looper = GatewayLooper(self.socket_map)
+
+    # Initialize metrics related
+    self.out_metrics = HeronCommunicator()
+    self.out_metrics.\
```

Comment:
can you break the line in the argument as compared to in the middle of invocation?